src/algorand/algorandTypes.ts

```diff
@@ -142,6 +142,12 @@ export const asMaybeCustomFee = asMaybe(
   { fee: undefined }
 )
 
+export const asMaybeAssetIndexLocation = asMaybe(
+  asObject({
+    assetIndex: asString
```

Comment:
Nothing prevents the index from being negative or floating point or whatever. Even the `getTokenId` checks don't handle these cases. Instead, steal this cleaner from edge-core-js and use it for `assetIndex`:

```typescript
export function asIntegerString(raw: unknown): string {
  const clean = asString(raw)
  if (!/^\d+$/.test(clean)) {
    throw new Error('Expected an integer string')
  }
  return clean
}
```